tests/h/search/conftest.py

```diff
@@ -16,6 +17,15 @@ def group_service(pyramid_config):
     return group_service
 
 
+@pytest.fixture(autouse=True)
+def moderation_service(pyramid_config):
+    svc = mock.create_autospec(AnnotationModerationService, spec_set=True, instance=True)
+    svc.all_hidden.return_value = []
+    svc.hidden.return_value = False
+    pyramid_config.register_service(svc, name='annotation_moderation')
+    return svc
```

Comment:
@sheetaluk Does this really need to be be an autouse fixture in `conftest.py`? That means that this fixture will be used for _every single test in the entire test suite_. There are two problems with this:

1. Running a fixture, especially an autospec, for every single test slows down the entire test run
2. Sharing the same fixture between a lot of tests couples those tests together. In the future if we want to change the behaviour of this fixture in order to write or fix one new test somewhere, that could cause any test anywhere in the test suite to break, because they're all sharing this fixture.

For these reasons we should keep the number of fixtures in `conftest.py` files as small as possible, and only add a fixture to a `conftest.py` if we really need to.

If this fixture is only need by the tests in `query_test.py`, for example, then this fixture can be moved into `query_test.py` and still have the `autouse=True` and then it will be automatically used for every test in that module but not for outside tests.

Some relevant notes on this:

* [Try not to over-reuse fixtures](https://www.seanh.cc/posts/advanced-fixtures#try-not-to-over-reuse-fixtures)

Reply:
@hmstepanek @robertknight @lyzadanger ^ It's worth every one taking note of this. Keeping our unit tests fast and decoupled from each other is really important.

Reply:
Ah, I didn't consider that. I suppose there are other test modules in here that don't need that fixture. Moving it back to the module an adding autouse sounds like the right approach.

Reply:
Leaving as is because there are 3 other tests `core_test`, `query_test` and `index_test` which are all in the same directory as `conftest.py`.